backend/app/api/src/helpers/FlagMomentCleanup.ts

```diff
@@ -0,0 +1,49 @@
+import { MemberResponsibilityRecord, Platform, Registration } from '@stamhoofd/models';
+import { SQL, SQLWhereExists, SQLWhereSign } from '@stamhoofd/sql';
+
+export class FlagMomentCleanup {
+    /**
+     * End functions of old members who have no active registration for the current period.
+     */
+    static async endFunctionsOfUsersWithoutRegistration() {
+        console.log('Start cleanup functions');
+        const responsibilitiesToEnd = await this.getActiveMemberResponsibilityRecordsForOrganizationWithoutRegistrationInCurrentPeriod();
+
+        const now = new Date();
+
+        await Promise.all(responsibilitiesToEnd.map(async (responsibility) => {
+            responsibility.endDate = now;
+            await responsibility.save();
+            console.log(`Ended responsibility with id ${responsibility.id}`);
+        }));
+    }
+
+    static async getActiveMemberResponsibilityRecordsForOrganizationWithoutRegistrationInCurrentPeriod() {
+        const currentPeriodId = (await Platform.getShared()).periodId;
+
+        const now = new Date();
+
+        return await MemberResponsibilityRecord.select()
+            .whereNot('organizationId', null)
+            .where(
+                SQL.where('startDate', SQLWhereSign.LessEqual, now)
+                    .or('startDate', null),
+            )
+            .where(
+                SQL.where('endDate', SQLWhereSign.GreaterEqual, now)
+                    .or('endDate', null),
+            )
+            .whereNot(
+                new SQLWhereExists(
```

Comment:
Dit is nog niet helemaal juist. Nu worden inschrijvingen voor activiteiten en wachtlijsten ook meegeteld.

De column 'waitingList' is niet meer relevant en is deprecated, is er puur nog om de oude data van Stamhoofd over te kunnen migreren naar de nieuwe data structuur, je hoeft/kan hier niet meer op filteren.

Je zal wrs een join moeten doen op 'groups' en filteren op groups.type.